### ISSUE TITLE
Problem with npm install

### ISSUE THREAD

**Issue by krivochenko** [angular/universal-starter#383]
Hi! I use Windows 10 with Docker Toolbox.

1. I clone with repo;
2. I added docker-compose.yml with following content:

```
version: '2'

services:
    nodejs:
        image: node:7.7.3
        volumes:
            - ./:/var/www/sources
        working_dir: /var/www/sources
        command: "npm install"          
```
3. run `docker-compose up`.

I get warnings:
```
nodejs_1  | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
nodejs_1  | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
nodejs_1  | npm WARN @ngtools/webpack@1.1.9 requires a peer of @angular/compiler-cli@2.2.3 but none was installed.
nodejs_1  | npm WARN @ngtools/webpack@1.1.9 requires a peer of @angular/core@2.2.3 but none was installed.
nodejs_1  | npm info ok
```

What I do wrong? Which information I can also provide? 

Thanks for help!

**Comment by krivochenko** [comment]
As I undestud, skipping optional dependency it's ok. I solved displaying of warnings about it by using --no-optional flag. But I steel geting other two warnings:

```
nodejs_1  | npm WARN @ngtools/webpack@1.1.9 requires a peer of @angular/compiler-cli@2.2.3 but none was installed.
nodejs_1  | npm WARN @ngtools/webpack@1.1.9 requires a peer of @angular/core@2.2.3 but none was installed.
```

**Comment by krivochenko** [comment]
I reproduced it on the server with Debian 8 and I get same result. I use same docker-image with Node 7.7.3 and NPM 4.1.2

If I be honest I was able to reproduce it just after I stoped other my containers. My container with Node was finishing with exit code 137 (OOM). 

```
nodejs_1  | npm info lifecycle universal-starter@2.0.0~preinstall: universal-starter@2.0.0
universal_nodejs_1 exited with code 137
```

What does universal-starter@2.0.0~preinstall do? What are the system requirements for this script?
